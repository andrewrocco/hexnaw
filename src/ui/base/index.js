export { default as colors } from './color/color';
export { default as colorVariables } from './color/colorVariables';
export { default as fonts } from './typography/fonts';
export { default as typography } from './typography/typography';
