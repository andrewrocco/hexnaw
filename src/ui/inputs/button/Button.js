import React from 'react';
import PropTypes from 'prop-types';

import * as Styled from './Button.styles';

export const Button = ({
  children,
  className,
  disabled,
  mode,
  onClick,
  text,
  type,
  ...others
}) => (
  <Styled.Button
    disabled={disabled}
    hasText={Boolean(text)}
    mode={mode}
    className={className}
    onClick={onClick}
    type={type}
    {...others}
  >
    {text || children}
  </Styled.Button>
);

const ButtonModes = ['inverse', 'none', 'primary'];

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  mode: PropTypes.oneOf(ButtonModes),
  onClick: PropTypes.func,
  text: PropTypes.string,
  type: PropTypes.string,
};

Button.defaultProps = {
  mode: 'primary',
  type: 'button',
};
