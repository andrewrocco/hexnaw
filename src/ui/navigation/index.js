export * from './header';
export * from './header_link';
export * from './header_logo';
