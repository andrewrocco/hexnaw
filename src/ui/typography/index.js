export * from './heading/Heading';
export * from './heading/Heading.styles';
export * from './text/Text';
