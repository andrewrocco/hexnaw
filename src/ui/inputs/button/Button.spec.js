import React from 'react';
import { Button } from 'ui/inputs';
import 'jest-styled-components';

import { modeStyles } from './Button.styles';

describe('Button', () => {
  context('when nothing specified', () => {
    it('renders correctly', () => {
      const subject = mount(<Button>Click it</Button>);
      expect(subject).toMatchSnapshot();
    });

    it('renders styles for "primary" mode', () => {
      const subject = mount(<Button>Click it</Button>);
      expect(subject).toHaveStyleRule('background', modeStyles.primary.enabled.background);
    });
  });

  context('when mode is specific', () => {
    it('renders mode-specific styles', () => {
      Object.keys(modeStyles).map((mode) => {
        const subject = mount(<Button mode={mode} text={mode} />);
        expect(subject).toHaveStyleRule('background', modeStyles[mode].enabled.background);
      });
    });
  });

  context('when disabled is true', () => {
    it('renders mode-specific disabled styles', () => {
      Object.keys(modeStyles).map((mode) => {
        const subject = mount(<Button disabled mode={mode} text={mode} />);
        expect(subject).toHaveStyleRule('background', modeStyles[mode].disabled.background);
      });
    });

    it('clicks do not fire the onClick function', () => {
      const onClick = jest.fn();
      const subject = mount(<Button onClick={onClick} disabled>Click Me</Button>);
      subject.simulate('click');
      expect(onClick).toHaveBeenCalledTimes(0);
    });
  });

  context('when you click it', () => {
    it('fires the onClick function', () => {
      const onClick = jest.fn();
      const subject = mount(<Button onClick={onClick}>Click Me</Button>);
      subject.simulate('click');
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
