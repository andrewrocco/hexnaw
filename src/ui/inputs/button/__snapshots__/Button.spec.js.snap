// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when nothing specified renders correctly 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #0201FE;
  border-color: #0201FE;
  border-style: solid;
  border-width: 1px;
  color: #FFFFFF;
  cursor: pointer;
  padding: 14px;
}

<Button
  mode="primary"
  type="button"
>
  <styled.button
    hasText={false}
    mode="primary"
    type="button"
  >
    <button
      className="c0"
      mode="primary"
      type="button"
    >
      Click it
    </button>
  </styled.button>
</Button>
`;
